extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -207,8 +211,16 @@ public OutputStream openStream() throws IOException
    */
 
   @Override
-  public String getStorageDir(DataSegment segment)
+  public String getStorageDir(DataSegment segment, boolean useUniquePath)
   {
+    // For HDFS, useUniquePath does not affect the directory tree but instead affects the filename, which is of the form
+    // '{partitionNum}_index.zip' without unique paths and '{partitionNum}_{UUID}_index.zip' with unique paths. Hence
+    // useUniquePath is ignored here and we expect it to be false.
+    Preconditions.checkArgument(
+        !useUniquePath,
+        "useUniquePath must be false for HdfsDataSegmentPusher.getStorageDir()"
```

Comment:
I don't think`HdfsDataSegmentPusher.getStorageDir()` is just called for Hadoop M/R, but it's only called from `HdfsDataSegmentPusher.push()` (which can be used by anything pushing to HDFS as a deep storage). But `HdfsDataSegmentPusher.push()` will always set this to false since any 'uniqueness' will be applied not to the directory but to the filename along with the shard (IIRC this was because of excessive directories in HDFS causing performance issues)

Reply:
Ah, I meant it's only called _directly_ by Hadoop M/R. Anyway this is great analysis - could you please include it as a comment?